acceptance/tests/options/external_facts.rb

```diff
@@ -1,67 +1,27 @@
-    custom_external_dir = get_user_fact_dir(agent['platform'], os_version)
-    ext = get_external_fact_script_extension(agent['platform'])
-
-    if agent['platform'] =~ /windows/
-      content = win_content
-    else
-      content = unix_content
-    end
-
-    step "Agent #{agent}: setup facts.d and custom external fact directories" do
-      on(agent, "mkdir -p '#{factsd}'")
-      on(agent, "mkdir -p '#{custom_external_dir}'")
-
-      step "Agent #{agent}: create executable external facts in facts.d and custom external fact dir" do
-        ext_fact_factsd     = File.join(factsd, "external_fact#{ext}")
-        ext_fact_custom_dir = File.join(custom_external_dir, "external_fact#{ext}")
-        create_remote_file(agent, ext_fact_factsd, content)
-        create_remote_file(agent, ext_fact_custom_dir, content)
-        on(agent, "chmod +x '#{ext_fact_factsd}' '#{ext_fact_custom_dir}'")
-
-        teardown do
-          on(agent, "rm -f '#{ext_fact_factsd}' '#{ext_fact_custom_dir}'")
-        end
-
-        step "--no-external-facts option should disable external facts" do
-          on(agent, facter("--no-external-facts external_fact")) do
-            assert_equal("", stdout.chomp, "Expected external fact to be disabled, but it resolved as #{stdout.chomp}")
-          end
-        end
-
-        step "--external-dir option should allow external facts to be resolved from a specific directory" do
-          on(agent, facter("--external-dir '#{custom_external_dir}' external_fact")) do
-            assert_equal("testvalue", stdout.chomp, "External fact output does not match expected output")
-          end
-        end
+    step "Agent #{agent}: create an external fact directory with an external fact" do
+      external_dir = agent.tmpdir('external_dir')
+      on(agent, "mkdir -p '#{external_dir}'")
+      ext = get_external_fact_script_extension(agent['platform'])
+      external_fact = File.join(external_dir, "external_fact#{ext}")
+      create_remote_file(agent, external_fact, external_fact_content(agent['platform'], 'single_fact', 'external_value'))
+      on(agent, "chmod +x '#{external_fact}'")
+
+      teardown do
+        on(agent, "rm -rf '#{external_dir}'")
+      end
 
-        step "--external-dir option should be able to be specified multiple times" do
-          on(agent, facter("--external-dir '#{custom_external_dir}' --external-dir 'a/fake/dir' external_fact")) do
-            assert_equal("testvalue", stdout.chomp, "External fact output does not match expected output")
-            assert_match(/skipping external facts for "a\/fake\/dir"/, stderr, "Did not search for second external fact directory")
-          end
+      step "Agent #{agent}: resolve a fact from each specified --external_dir option" do
+        on(agent, facter("--external-dir #{external_dir} single_fact")) do
+          assert_equal("external_value", stdout.chomp, "External fact output does not match expected output #{stdout.chomp}")
```

Comment:
Having the actual result after the phrase 'expected output' is not intuitive to me. I would misinterpret this value as the expected value.